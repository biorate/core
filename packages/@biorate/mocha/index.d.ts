import {
  ExecutionModifierDecorator,
  ExecutionOptionDecorator,
  ParameterisedTestDecorator,
  SuiteDecorator,
  TestDecorator,
} from './src/core';

declare module '@biorate/mocha' {
  export const context: unique symbol;
  export const suite: SuiteDecorator;
  export const test: TestDecorator;
  export const slow: ExecutionOptionDecorator;
  export const timeout: ExecutionOptionDecorator;
  export const retries: ExecutionOptionDecorator;
  export const pending: ExecutionModifierDecorator;
  export const only: ExecutionModifierDecorator;
  export const skip: ExecutionModifierDecorator;
  export const params: ParameterisedTestDecorator;
}
