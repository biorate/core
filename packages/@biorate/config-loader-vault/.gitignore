# compiled output
dist
node_modules
cache
keys

# Logs
logs
*.log
npm-debug.log*
yarn-debug.log*
yarn-error.log*
lerna-debug.log*

# Diagnostic reports (https://nodejs.org/api/report.html)
report.[0-9]*.[0-9]*.[0-9]*.[0-9]*.json

# OS
.DS_Store

# Coverage directory used by tools like istanbul
coverage
*.lcov

# nyc test coverage
.nyc_output

# Optional npm cache directory
.npm

# Optional eslint cache
.eslintcache

# IDEs and editors
.idea
.project
.classpath
.c9/
*.launch
.settings/
*.sublime-workspace

# IDE - VSCode
.vscode/*
!.vscode/settings.json
!.vscode/tasks.json
!.vscode/launch.json
!.vscode/extensions.json

# Package lock
package-lock.json
