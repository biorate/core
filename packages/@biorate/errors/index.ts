import { format } from 'util';

/**
 * @description
 * This module provides a base class for creating error classes with error code (class name),
 * message template, meta data support out of the box.
 *
 * @example
 * ```ts
 * import { BaseError } from '@biorate/errors';
 *
 * export class MyAwesomeError extends BaseError {
 *   constructor(args?: any[], meta?: any) {
 *     super(`Oops... Some error happen, at [%s], in [%s]`, args, meta);
 *   }
 * }
 *
 * const e = new MyAwesomeError([new Date(), 'core'], { hello: 'world!' });
 *
 * console.log(e.meta); // { hello: 'world!' }
 * console.log(e.code); // MyAwesomeError
 *
 * throw e;
 *           ^
 * MyAwesomeError: Oops... Some error happen, at [2021-05-13T09:19:22.511Z], in [core]
 *     at Object.<anonymous> (..core/packages/@biorate/errors/index.ts:28:11)
 *     at Module._compile (internal/modules/cjs/loader.js:1138:30)
 * ```
 */
export class BaseError extends Error {
  #meta: any = null;

  public constructor(
    message: string,
    args?: unknown[],
    meta?: unknown,
    ...options: unknown[]
  ) {
    // @ts-ignore: { cause } property added in 16.9.0
    super(format(message, ...(args || [])), ...options);
    this.#meta = meta;
    this.name = this.constructor.name;
    Error.captureStackTrace(this, this.constructor);
  }

  public get meta() {
    return this.#meta;
  }

  public get code() {
    return this.constructor.name;
  }
}
