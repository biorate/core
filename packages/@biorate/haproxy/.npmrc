registry=https://art.lmru.tech/api/npm/npm-payment/
package-lock=false
