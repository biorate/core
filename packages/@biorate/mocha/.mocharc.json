{
  "require": "source-map-support/register,ts-node/register",
  "spec": "tests/**/*.spec.ts",
  "reporter": "mocha-multi-reporters",
  "reporter-option": "configFile=.reporters.json",
  "exit": true,
  "parallel": false
}
