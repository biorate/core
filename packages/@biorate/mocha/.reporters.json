{
  "reporterEnabled": "allure-mocha,spec",
  "allureMochaReporterOptions": {
    "resultsDir": "./allure-results"
  }
}
