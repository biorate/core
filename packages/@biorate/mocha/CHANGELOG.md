# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.52.0](https://github.com/biorate/core/compare/v1.51.1...v1.52.0) (2024-01-10)

### Features

- **mocha:** Spec abstract class added ([60d7596](https://github.com/biorate/core/commit/60d7596ff7873c3f0590d9afe4ed1d9b4ff333e6))

# [1.51.0](https://github.com/biorate/core/compare/v1.50.0...v1.51.0) (2023-12-11)

### Features

- **playwright:** types ([3c6e4bb](https://github.com/biorate/core/commit/3c6e4bb7efe379c047a1b21c7bd821a3bc5258f0))

# [1.49.0](https://github.com/biorate/core/compare/v1.48.0...v1.49.0) (2023-12-11)

**Note:** Version bump only for package @biorate/mocha

# [1.48.0](https://github.com/biorate/core/compare/v1.47.0...v1.48.0) (2023-12-07)

### Features

- **mocha:** parallel tests ([e219566](https://github.com/biorate/core/commit/e219566a5d9b566f94e53fb99227146c2df90da6))

# [1.47.0](https://github.com/biorate/core/compare/v1.46.0...v1.47.0) (2023-12-07)

### Features

- **mocha:** parallel tests ([b8626be](https://github.com/biorate/core/commit/b8626be52d2d15674dbae1a634d142f4ee85d38d))

# [1.46.0](https://github.com/biorate/core/compare/v1.45.1...v1.46.0) (2023-12-07)

### Features

- **mocha:** parallel tests ([b956faf](https://github.com/biorate/core/commit/b956faf0ba512a589adfdc697b7190a3ffd23542))

## [1.45.1](https://github.com/biorate/core/compare/v1.45.0...v1.45.1) (2023-11-30)

### Bug Fixes

- **mocha:** src add into npm ([edffa4d](https://github.com/biorate/core/commit/edffa4dd1f911ee865359d6d80147fc902d44faf))

# [1.45.0](https://github.com/biorate/core/compare/v1.44.1...v1.45.0) (2023-11-30)

### Features

- **mocha:** module added ([af41dca](https://github.com/biorate/core/commit/af41dca35e6b2f99841bf1d64b2282dbbfde7763))
- **mocha:** module added ([954a82d](https://github.com/biorate/core/commit/954a82d6b411371ecca81bad855331200e608f80))
