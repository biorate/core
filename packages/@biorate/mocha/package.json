{
  "name": "@biorate/mocha",
  "version": "1.46.0",
  "description": "Mocha OOP tests based on @testdeck/core",
  "main": "dist",
  "scripts": {
    "build": "npx tsc -p ./tsconfig.build.json --outDir ./dist",
    "test": "npx nyc --reporter=lcov --reporter=text-summary -- npx mocha --exit -r ts-node/register tests/**/*.spec.ts",
    "prepublishOnly": "npm run build"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/biorate/core.git",
    "directory": "packages/@biorate/mocha"
  },
  "publishConfig": {
    "access": "public"
  },
  "homepage": "https://biorate.github.io/core",
  "keywords": [
    "mocha",
    "OOP",
    "parallel",
    "@testdeck/core"
  ],
  "author": "llevkin",
  "license": "MIT",
  "dependencies": {
    "mocha": "^10.2.0"
  }
}
