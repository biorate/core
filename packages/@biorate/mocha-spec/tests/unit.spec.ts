import { suite, test } from '@biorate/mocha';
import { Spec, unitTests } from './__mocks__';

@suite('@biorate/mocha-spec - unit')
class MochaUnitSpec extends Spec {
  @test('all-expects-positive')
  protected async allExpectsPositive() {
    await this.unit({
      context: unitTests,
      method: 'increment',
      args: [3],
      expects: {
        context: true,
        args: true,
        return: true,
      },
    });
  }

  @test('all-expects-negative')
  protected async allExpectsNegative() {
    await this.unit({
      context: unitTests,
      method: 'throw',
      args: [],
      expects: {
        context: true,
        args: true,
        return: true,
      },
      catch: (e: Error) => e.message.includes('Test error'),
    });
  }

  @test('custom-expects-positive')
  protected async customExpectsPositive() {
    await this.unit({
      context: unitTests,
      method: 'passthrow',
      args: [3, 2],
      expects: {
        context: ['dec'],
        args: ['[0]', '[1]'],
        return: ['a'],
      },
    });
  }
}
