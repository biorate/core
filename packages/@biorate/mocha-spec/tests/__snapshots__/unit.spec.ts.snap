// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@biorate/mocha-spec - unit all-expects-negative 1`] = `Array []`;

exports[`@biorate/mocha-spec - unit all-expects-negative 2`] = `
UnitTests {
  "dec": 1,
  "inc": 4,
}
`;

exports[`@biorate/mocha-spec - unit all-expects-positive 1`] = `
Array [
  3,
]
`;

exports[`@biorate/mocha-spec - unit all-expects-positive 2`] = `
UnitTests {
  "dec": 1,
  "inc": 1,
}
`;

exports[`@biorate/mocha-spec - unit all-expects-positive 3`] = `
UnitTests {
  "dec": 1,
  "inc": 4,
}
`;

exports[`@biorate/mocha-spec - unit all-expects-positive 4`] = `
Array [
  3,
]
`;

exports[`@biorate/mocha-spec - unit all-expects-positive 5`] = `4`;

exports[`@biorate/mocha-spec - unit custom-expects-positive 1`] = `
Array [
  3,
  2,
]
`;

exports[`@biorate/mocha-spec - unit custom-expects-positive 2`] = `1`;

exports[`@biorate/mocha-spec - unit custom-expects-positive 3`] = `1`;

exports[`@biorate/mocha-spec - unit custom-expects-positive 4`] = `3`;

exports[`@biorate/mocha-spec - unit custom-expects-positive 5`] = `2`;

exports[`@biorate/mocha-spec - unit custom-expects-positive 6`] = `3`;
